File: April-18/03-04-18/python/02-Once-upon-a-time-in-Time-Land.py
for _ in range( int(input()) ):
    n, k = map(int, input().split())
    ar = [int(x) for  x in input().split()]
    dp = ar
    
    maxi = 0
    
    for i in range(0, n):
        if(i- (k+1) >= 0):
            dp[i] = max(maxi, dp[i] + dp[i-(k+1)])
        else:
            dp[i] = max(maxi, dp[i])
        if(dp[i] > maxi):
            maxi = dp[i]
    print(maxi)
    

File: April-18/03-04-18/java/02-Once-upon-a-time-in-Time-Land.java
import java.util.*;

class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner input = new Scanner(System.in);
        int t = input.nextInt();
        while(t-- > 0)
        {
            int n, k;
            n = input.nextInt();
            k = input.nextInt();
            long dp[] = new long[n+1];
            
            long maxi = 0;
            for(int i = 1; i <= n; i++)
            {
                dp[i] = input.nextLong();
            }
            
            for(int i = 1; i <= n; i++)
            {
                if(i - (k+1) > 0)
                    dp[i] = Math.max(maxi , dp[i] + dp[i-(k+1)]);
                else
                    dp[i] = Math.max(dp[i], maxi);
                if(dp[i] > maxi)
                    maxi = dp[i];
                    
            }
            
            System.out.println(maxi);
            
        }
        
    }
}
